src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -265,16 +391,18 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
                         ExpandedCompletionBehavior = ExpandedCompletionMode.ExpandedItemsOnly
                     };
 
-                    return await GetCompletionContextWorkerAsync(document, session, intialTrigger, initialTriggerLocation, options, cancellationToken).ConfigureAwait(false);
+                    var (context, completionList) = await GetCompletionContextWorkerAsync(document, intialTrigger, initialTriggerLocation, options, cancellationToken).ConfigureAwait(false);
+                    AddPropertiesToSession(session, completionList, triggerLocation: null);
+
+                    return context;
                 }
             }
 
-            return AsyncCompletionData.CompletionContext.Empty;
+            return VSCompletionContext.Empty;
         }
 
-        private async Task<AsyncCompletionData.CompletionContext> GetCompletionContextWorkerAsync(
+        private async Task<(VSCompletionContext, CompletionList)> GetCompletionContextWorkerAsync(
```

Comment:
i'm a bit lost as to why we have two pieces of information here.  why does the VSCompletionContext not hold onto the completion list?

Reply:
First of all, we need a few piece of information from CompletionList to save in the session property bag, which is why we are returning it. That said, the CompletionList (or those specific info we need) could be saved in VSCompletionContext's property bag instead, but I didn't want to do that (yet another property bag!)

Is that what you were asking?